### PR TITLE
bugfix(form field icon): icon centralized in input field

### PR DESCRIPTION
**Pull request by beaolivei**
The icon is centralized now. This pull request also starts adapting the input to its new fill style.

### CODE REVIEW

File: themes/angular-theme/lib/input/_input-base.scss [hunk 1]
```diff
@@ -1,4 +1,13 @@
-.mat-form-field-prefix,
-.mat-form-field-suffix {
-  align-self: flex-end;
+.mat-form-field-prefix {
+  .mat-icon {
+    margin-right: 12px;
```
Comment (rbordeanu) [inline themes/angular-theme/lib/input/_input-base.scss]: We shouldn't impose a margin on the mat-icon prefix.
Let the implementer choose if he wants one.

File: themes/angular-theme/lib/input/_input-base.scss [hunk 2]
```diff
@@ -1,4 +1,13 @@
...
+  }
+}
+
+.mat-form-field-appearance-legacy {
+  .mat-form-field-wrapper .mat-form-field-infix {
+    .mat-input-element {
+      padding: 4px 0;
```
Comment (rbordeanu) [inline themes/angular-theme/lib/input/_input-base.scss]: This causes the input text to not be on the same alignment with the placeholder:
Screenshot 2020-03-09 at 00 47 16
Screenshot 2020-03-09 at 00 47 30

File: themes/angular-theme/lib/input/_input-base.scss [hunk 3]
```diff
@@ -1,4 +1,13 @@
-.mat-form-field-prefix,
-.mat-form-field-suffix {
-  align-self: flex-end;
```
Comment (rbordeanu) [inline themes/angular-theme/lib/input/_input-base.scss]: If you remove the 4px added padding (see the other comment), we should keep this setting, but only for legacy appearance.

File: themes/angular-theme/lib/input/_input-base.scss (hunk 2 above)
Comment (beaolivei) [reply themes/angular-theme/lib/input/_input-base.scss]: this was removed as well. Thanks for pointing it out 👍

File: themes/angular-theme/lib/input/_input-base.scss (hunk 3 above)
Comment (beaolivei) [reply themes/angular-theme/lib/input/_input-base.scss]: Absolutely right 👍

File: themes/angular-theme/lib/icon/_icon-base.scss [hunk 4]
```diff
@@ -6,10 +6,10 @@ $icon-size-base: $uxg-spacing-4;
 $icon-size-dense: 18px;
 
 @mixin icon-size($icon-size) {
-  width: $icon-size;
-  height: $icon-size;
-  font-size: $icon-size;
-  line-height: $icon-size;
+  width: $icon-size !important;
```
[the remark this answers is not in the document]
Comment (rbordeanu) [reply themes/angular-theme/lib/icon/_icon-base.scss]: it's ok, they were there from the beginning :) not a recommended pattern but sometimes it's needed...

**Pull request merged**